commons-statistics-descriptive/src/test/java/org/apache/commons/statistics/descriptive/MaxTest.java

```diff
@@ -0,0 +1,188 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.statistics.descriptive;
+
+import java.util.Arrays;
+import java.util.function.DoubleSupplier;
+import java.util.stream.Stream;
+import org.apache.commons.rng.UniformRandomProvider;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Test for {@link Max}.
+ */
+final class MaxTest {
+
+    @Test
+    void testEmpty() {
+        Max max = Max.create();
+        Assertions.assertEquals(Double.NEGATIVE_INFINITY, max.getAsDouble());
+    }
+
+    @Test
+    void testIncrement() {
+        // Test the max after each incremental update
+        // First parameter of testArray is the value that would be added
+        // Second parameter of testArray is the max we expect after adding the value
+        double[][] testArray = {
+            {1729.22, 1729.22},
+            {2520.35, 2520.35},
+            {2010.87, 2520.35},
+            {100000000.1, 100000000.1},
+            {+0.0, 100000000.1},
+            {-0.0, 100000000.1},
+            {Double.MAX_VALUE, Double.MAX_VALUE},
+            {Double.POSITIVE_INFINITY, Double.POSITIVE_INFINITY},
+            {Double.NEGATIVE_INFINITY, Double.POSITIVE_INFINITY}
+        };
+
+        Max stat = Max.create();
+        for (final double[] valueAndExpected: testArray) {
+            final double value = valueAndExpected[0];
+            final double expected = valueAndExpected[1];
+            stat.accept(value);
+            Assertions.assertEquals(expected, stat.getAsDouble());
+        }
+    }
+
+    @Test
+    void testNaN() {
+        // Test non-nan values cannot revert a NaN
+        double[] testArray = {Double.NaN, +0.0d, -0.0d, Double.POSITIVE_INFINITY, Double.NEGATIVE_INFINITY};
+        Max stat = Max.create();
+        for (final double x : testArray) {
+            stat.accept(x);
+            Assertions.assertEquals(Double.NaN, stat.getAsDouble());
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void testMax(double[] values, double expected) {
+        Max stat = Max.create();
+        Arrays.stream(values).forEach(stat);
+        double actual = stat.getAsDouble();
+        Assertions.assertEquals(expected, actual, "max");
+        Assertions.assertEquals(expected, Max.of(values).getAsDouble(), "max");
```

Comment:
The assertion message here is not helpful to distinguish this case. You could use "of(values)". Just change the test data so the test fails and you can see how JUnit will output the message. This should be helpful to diagnose the problem with the code.